python/plugins/processing/gui/AlgorithmDialog.py

```diff
@@ -154,7 +155,19 @@ def getParameterValues(self):
                     if self.mainWidget().checkBoxes[param.name()].isChecked():
                         dest_project = QgsProject.instance()
 
-                value = self.mainWidget().outputWidgets[param.name()].getValue()
+                widget = self.mainWidget().outputWidgets[param.name()]
+                value = widget.getValue()
+                outputPath, _ = value.sink.valueAsString(self.context.expressionContext())
```

Comment:
I'd like to see this whole block moved to c++, and sit within QgsProcessingProvider. E.g. something like:

    /**
     * Returns true if the specified \a outputValue is of a supported file format for the given destination \a parameter.
    */
    bool isSupportedFormat( const QVariant& outputValue, const QgsProcessingDestinationParameter* parameter, QgsProcessingContext& context ) const;

Reply:
Right. If you add that to the provider C++ class, let me know and I will adapt the Python part accordingly.